### PR TITLE
Seo improvements

### PR DESCRIPTION
**Pull request by elbotho**
### Added

- Improved title in entities (e.g. automated " - lernen mit Serlo" suffix) ([#678](https://github.com/serlo-org/athene2/pull/678))

### Changed

- Show a subject-specific Open Graph meta image (e.g. for Facebook previews) ([#678](https://github.com/serlo-org/athene2/pull/678))
- Expose a tenant-spefiic Open Search description file ([#678](https://github.com/serlo-org/athene2/pull/678))
---

For #436, #650 and #654 and also adding canonical link tag.

@inyono needs assets and new images in assets (see question in commits)

### CODE REVIEW

**Comment by Knorrke** [comment]
@inyono I've structured the code as discussed and rebased it onto the current development branch. There's still an issue though: The Meta-Description field is ignored now. Didn't get it to work atm, and probably won't look at it again until monday. Feel free to take a look, if you want.

File: src/module/Normalizer/src/Normalizer/View/Helper/Normalize.php [hunk 1]
```diff
@@ -39,33 +42,100 @@ public function headMeta($object)
...
+
+            case 'biologie':
+            case 'biology':
+                $fileName = 'meta_serlo_bio.png';
+                break;
+
+            default:
+                break;
+        }
+        return 'https://de.serlo.org/assets/images/'.$fileName;
```
Comment (inyono) [inline src/module/Normalizer/src/Normalizer/View/Helper/Normalize.php]: @inyono: FIXME, probably using assets.serlo.org?

File: src/module/Normalizer/src/Normalizer/View/Helper/Normalize.php [hunk 2]
```diff
@@ -53,33 +56,101 @@ public function headMeta($object)
...
+                ->setPartial('layout/navigation/partial/active-subject')
+                ->setOnlyActiveBranch(true)
+                ->setMinDepth(0)
+                ->setMaxDepth(0)
+                ->render()
+        )));
+
+        switch ($subject) {
+            case 'mathematik':
+            case 'math':
```
Comment (inyono) [inline src/module/Normalizer/src/Normalizer/View/Helper/Normalize.php]: IMHO: either we add English versions of the images or remove the behaviour for `en.serlo.org` for now.

File: src/module/Normalizer/src/Normalizer/Adapter/AbstractAdapter.php [hunk 3]
```diff
@@ -25,11 +25,18 @@
 use Normalizer\Entity\Normalized;
 use DateTime;
 use Normalizer\Exception\RuntimeException;
+use Zend\I18n\Translator\TranslatorInterface;
```
Comment (inyono) [inline src/module/Normalizer/src/Normalizer/Adapter/AbstractAdapter.php]: use `Zend\I18n\Translator\TranslatorAwareInterface` instead

File: src/module/Normalizer/src/Normalizer/Adapter/AdapterInterface.php [hunk 4]
```diff
@@ -23,6 +23,7 @@
 namespace Normalizer\Adapter;
 
 use Normalizer\Entity\NormalizedInterface;
+use Zend\I18n\Translator\TranslatorInterface;
```
Comment (inyono) [inline src/module/Normalizer/src/Normalizer/Adapter/AdapterInterface.php]: use `Zend\I18n\Translator\TranslatorAwareInterface` instead

File: src/module/Normalizer/src/Normalizer/Factory/NormalizeHelperFactory.php [hunk 5]
```diff
@@ -38,7 +38,10 @@ public function createService(ServiceLocatorInterface $serviceLocator)
     {
         $normalize  = new Normalize();
         $normalizer = $serviceLocator->getServiceLocator()->get('Normalizer\Normalizer');
+        $instanceManager = $serviceLocator->getServiceLocator()->get('Instance\Manager\InstanceManager');
```
Comment (inyono) [inline src/module/Normalizer/src/Normalizer/Factory/NormalizeHelperFactory.php]: added by @inyono: pass `InstanceManager` to `Normalizer` so we get a tenant-aware open search

File: src/module/Normalizer/src/Normalizer/Normalizer.php [hunk 6]
```diff
@@ -24,6 +24,7 @@
 
 use Normalizer\Adapter\AdapterPluginManager;
 use Zend\Cache\Storage\StorageInterface;
+use Zend\I18n\Translator\TranslatorInterface;
```
Comment (inyono) [inline src/module/Normalizer/src/Normalizer/Normalizer.php]: use `TranslatorAwareInterface` instead

File: src/module/Normalizer/src/Normalizer/View/Helper/Normalize.php [hunk 7]
```diff
@@ -43,43 +48,152 @@ public function __invoke($object = null)
...
+
         return $this;
     }
 
-    public function getMetaImage($object)
+    private function appendOpenSearchMeta($object)
+    {
+        $lang = $this->getSubdomain();
+
+        if ($lang === 'de' || $lang === 'en') {
```
Comment (inyono) [inline src/module/Normalizer/src/Normalizer/View/Helper/Normalize.php]: added by @inyono: tenant-aware open search

File: src/module/Normalizer/src/Normalizer/View/Helper/Normalize.php (hunk 2 above)
Comment (inyono) [reply src/module/Normalizer/src/Normalizer/View/Helper/Normalize.php]: @Knorrke: I removed the handling for subjects on en.serlo.org for now (i.e. the fallback image is used instead). Are you fine with that?

Comment (Knorrke) [reply src/module/Normalizer/src/Normalizer/View/Helper/Normalize.php]: Yeah, that's fine I guess

**Comment by Knorrke** [comment]
Ok, I just don't know how this should work... The meta description is only text, but the fallback is markdown and needs to be rendered. So I probably want to use `toPreview($object)` in that case. But I can't access that in the adapters...

**Comment by Knorrke** [comment]
It's done now from my side, thanks again for the help 👍

**Pull request merged**